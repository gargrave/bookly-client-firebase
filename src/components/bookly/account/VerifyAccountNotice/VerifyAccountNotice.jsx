// @flow
import React from 'react';
import { bool, func } from 'prop-types';

import Button from '../../../common/Button';
import Card from '../../../common/Card';
import CardSpacer from '../../../common/Card/CardSpacer';
import CardTextLine from '../../../common/Card/CardTextLine';

type Props = {
  emailHasBeenSent: boolean,
  onSendLinkClick: Function,
};

const buttonText = (emailHasBeenSent) =>
  emailHasBeenSent ? 'Verification Link Has Been Sent!' : 'Resend Verification Link';

const buttonClass = (emailHasBeenSent) =>
  emailHasBeenSent ? 'info' : 'warning';

const VerifyAccountNotice = ({
  emailHasBeenSent,
  onSendLinkClick,
}: Props) => {
  return (
    <Card
      classes={['detail-card', 'verify-account-card']}
      hoverable={false}
    >
      <CardTextLine
        bold={true}
        text={'Your email address has not been verified!'}
      />
      <CardTextLine
        text={'Your must verify your email address before your account will be active.'}
      />
      <CardSpacer />

      <Button
        onClick={onSendLinkClick}
        text={buttonText(emailHasBeenSent)}
        type={buttonClass(emailHasBeenSent)}
      />
    </Card>
  );
};

VerifyAccountNotice.propTypes = {
  emailHasBeenSent: bool.isRequired,
  onSendLinkClick: func.isRequired,
};

export default VerifyAccountNotice;
