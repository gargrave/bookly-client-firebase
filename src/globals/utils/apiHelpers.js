// @flow
import type { FbDocRef } from '../../modules/core/flowtypes';

async function getDocRef(
  db: any,
  table: string,
  id: string,
): FbDocRef {
  return db.collection(table).doc(id);
}

export {
  getDocRef,
};
