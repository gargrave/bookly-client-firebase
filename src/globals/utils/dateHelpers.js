export const toUnixTimestamp = (date, defaultValue) => {
  if (!date) {
    if (defaultValue === null || defaultValue === undefined) {
      return Date.now();
    }
    return defaultValue;
  }

  if (date._isAMomentObject) {
    return date.unix();
  }

  if (typeof date === 'number') {
    return date;
  }

  return defaultValue;
};

export const toUnixMs = (date) =>
  (date && date * 1000) || 0;
