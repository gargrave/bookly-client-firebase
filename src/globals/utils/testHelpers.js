import React from 'react';
import { shallow } from 'enzyme';

export class ComponentBuilder {
  constructor(component, defaultProps) {
    this.component = component;
    this.defaultProps = defaultProps;
  }

  shallowGetComponent(extraProps = {}) {
    const props = {
      ...this.defaultProps,
      ...extraProps,
    };
    return shallow(<this.component {...props} />);
  }
}
