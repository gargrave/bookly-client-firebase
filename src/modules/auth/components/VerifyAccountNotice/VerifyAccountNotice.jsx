// @flow
import React from 'react';
import { bool, func } from 'prop-types';

import Button from '../../../common/components/Button/Button';
import Card from '../../../common/components/Card/Card';
import CardSpacer from '../../../common/components/Card/CardSpacer/CardSpacer';
import CardTextList from '../../../common/components/Card/CardTextList/CardTextList';

type Props = {
  emailHasBeenSent: boolean,
  onSendLinkClick: Function,
};

const buttonText = (emailHasBeenSent) =>
  emailHasBeenSent ? 'Verification Link Has Been Sent!' : 'Resend Verification Link';

const buttonClass = (emailHasBeenSent) =>
  emailHasBeenSent ? 'info' : 'warning';

const textList = [
  { title: '', value: 'You must verify your email address before your account will be fully activated.' },
  {
    title: '',
    value: 'You should have received a verification link at your email address when you registered, ' +
      'but if you need a new link, you can request one below.',
  },
  { title: 'NOTE', value: 'After verifying your email, you MUST log out and log back in one time for the activation to take effect.' },
];

const VerifyAccountNotice = ({
  emailHasBeenSent,
  onSendLinkClick,
}: Props) => {
  return (
    <Card
      classes={['detail-card', 'verify-account-card']}
      hoverable={false}
      title={'Your email address has not been verified!'}
    >
      <CardSpacer />
      <CardTextList textList={textList} />
      <CardSpacer />

      <Button
        disabled={emailHasBeenSent}
        onClick={onSendLinkClick}
        text={buttonText(emailHasBeenSent)}
        type={buttonClass(emailHasBeenSent)}
      />
    </Card>
  );
};

VerifyAccountNotice.propTypes = {
  emailHasBeenSent: bool.isRequired,
  onSendLinkClick: func.isRequired,
};

export default VerifyAccountNotice;
