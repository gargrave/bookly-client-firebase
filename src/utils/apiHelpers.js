// @flow
import type { FbDocRef } from '../constants/flowtypes';

async function getDocRef(
  db: any,
  table: string,
  id: string,
): FbDocRef {
  return db.collection(table).doc(id);
}

export {
  getDocRef,
};
