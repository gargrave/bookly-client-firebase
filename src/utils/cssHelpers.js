// @flow
export function buildClass(...classArgs: any): string {
  const classList = [];
  classArgs.forEach((c) => {
    // for strings, we can add the immediately
    if (typeof c === 'string') {
      classList.push(c);
    }

    // for objects, assume there are multiple values to test,
    // and add each one to the list if its condition passes
    if (typeof c === 'object') {
      Object.keys(c).forEach((key) => {
        if (c[key]) {
          classList.push(key);
        }
      });
    }
  });
  return classList.join(' ');
}
