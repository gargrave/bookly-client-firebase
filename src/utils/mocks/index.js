import authorMocks from './data/authors';
import userMocks from './data/users';

export {
  authorMocks,
  userMocks,
};
