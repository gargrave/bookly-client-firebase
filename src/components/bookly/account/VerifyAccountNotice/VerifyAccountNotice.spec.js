import { ComponentBuilder } from '../../../../globals/utils/testHelpers';

import VerifyAccountNotice from './VerifyAccountNotice';

const defaultProps = {
  onSendLinkClick: jest.fn(),
};

const builder = new ComponentBuilder(
  VerifyAccountNotice,
  defaultProps,
);

describe('VerifyAccountNotice', () => {
  let component;

  it('matches the snapshot', () => {
    component = builder.shallowGetComponent();
    expect(component).toMatchSnapshot();
  });

  describe('basic rendering', () => {
    it('renders correctly', () => {
      component = builder.mountComponent();
      expect(component.find('.bookly-detail-card')).toHaveLength(1);
      expect(component.find('.bookly-verify-account-card')).toHaveLength(1);
    });

    it('should not be hoverable', () => {
      component = builder.mountComponent();
      expect(component.find('.bookly-detail-card--hoverable')).toHaveLength(0);
    });
  });
});
