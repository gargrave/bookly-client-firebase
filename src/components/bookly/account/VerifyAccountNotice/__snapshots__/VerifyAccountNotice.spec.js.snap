// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyAccountNotice matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VerifyAccountNotice
    emailHasBeenSent={false}
    onSendLinkClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CardTextLine
          bold={true}
          text="Your email address has not been verified!"
        />,
        <CardTextLine
          text="Your must verify your email address before your account will be active."
        />,
        <CardSpacer />,
        <Button
          onClick={[Function]}
          text="Resend Verification Link"
          type="warning"
        />,
      ],
      "classes": Array [
        "detail-card",
        "verify-account-card",
      ],
      "hoverable": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bold": true,
          "text": "Your email address has not been verified!",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "text": "Your must verify your email address before your account will be active.",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "text": "Resend Verification Link",
          "type": "warning",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardTextLine
            bold={true}
            text="Your email address has not been verified!"
          />,
          <CardTextLine
            text="Your must verify your email address before your account will be active."
          />,
          <CardSpacer />,
          <Button
            onClick={[Function]}
            text="Resend Verification Link"
            type="warning"
          />,
        ],
        "classes": Array [
          "detail-card",
          "verify-account-card",
        ],
        "hoverable": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bold": true,
            "text": "Your email address has not been verified!",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "text": "Your must verify your email address before your account will be active.",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "text": "Resend Verification Link",
            "type": "warning",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
